scripts/upload-ami

```diff
@@ -26,6 +31,39 @@ ALIYUN = 'aliyun'
 
 cloud = None
 
+# For *some* regions in Aliyun you have to use a specific endpoint,
+# which is weird
+def get_aliyun_endpoint(region):
+    endpoint = f'ecs.{region}.aliyuncs.com'
+    if region in ['us-east-1', 'us-west-1', 'cn-qingdao',
+                  'cn-beijing', 'cn-hangzhou', 'cn-shanghai',
+                  'cn-shanghai', 'cn-hongkong', 'ap-southeast-1']:
+        endpoint = 'ecs.aliyuncs.com'
+
+    return endpoint
+
+
+# Find the image_id in meta.json given a build_id, region, arch
+def find_image_id(cloud_id, region_id, build_id, arch):
+    with open(f"builds/{build_id}/{arch}/meta.json") as meta_f:
+        meta = json.load(meta_f)
+    image_id = None
+
+    if cloud_id == AWS:
+        for region in meta['amis']:
+            if region['name'] == region_id:
+                image_id = region['hvm']
+
+    if cloud_id == ALIYUN:
```

Comment:
Minor nit, I think this should be `else if` since the cases are mutually exclusive.